.circleci/config.yml

```diff
@@ -400,101 +444,20 @@ commands:
-      - run:
-          name: Install Node.js and yarn
-          command: |
-            export DEBIAN_FRONTEND=noninteractive
-            nvm install
-            node --version
-
-            curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
-            echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
-            sudo apt-get update && sudo apt-get --yes install --no-install-recommends yarn=<< parameters.yarn_version >>
-            yarn --version
-      - when:
-          condition:
-            equal: ["testcafe", "<< parameters.test_runner >>"]
-          steps:
-            - run:
-                name: Install browsers
-                command: |
-                  set -e
-                  curl --silent --show-error --location --fail --retry 3 --output /tmp/google-chrome-stable_current_amd64.deb https://voltti-public.s3.eu-west-1.amazonaws.com/google-chrome-stable_current_amd64.deb \
-                    && (sudo dpkg -i /tmp/google-chrome-stable_current_amd64.deb || sudo apt-get -fy install)  \
-                    && rm -rf /tmp/google-chrome-stable_current_amd64.deb \
-                    && sudo sed -i 's|HERE/chrome"|HERE/chrome" --disable-setuid-sandbox --no-sandbox|g' \
-                        "/opt/google/chrome/google-chrome" \
-                    && google-chrome --version
-                  CHROME_VERSION="$(google-chrome --version)" \
-                    && export CHROMEDRIVER_RELEASE="$(echo $CHROME_VERSION | sed 's/^Google Chrome //')" && export CHROMEDRIVER_RELEASE=${CHROMEDRIVER_RELEASE%%.*} \
-                    && CHROMEDRIVER_VERSION=$(curl --silent --show-error --location --fail --retry 4 --retry-delay 5 http://chromedriver.storage.googleapis.com/LATEST_RELEASE_${CHROMEDRIVER_RELEASE}) \
-                    && curl --silent --show-error --location --fail --retry 4 --retry-delay 5 --output /tmp/chromedriver_linux64.zip "http://chromedriver.storage.googleapis.com/$CHROMEDRIVER_VERSION/chromedriver_linux64.zip" \
-                    && cd /tmp \
-                    && unzip chromedriver_linux64.zip \
-                    && rm -rf chromedriver_linux64.zip \
-                    && sudo mv chromedriver /usr/local/bin/chromedriver \
-                    && sudo chmod +x /usr/local/bin/chromedriver \
-                    && chromedriver --version
-      - when:
-          condition:
-            equal: ["playwright", "<< parameters.test_runner >>"]
-          steps:
-            - run:
-                name: Install Playwright dependencies
-                command: |
-                  sudo apt-get update && sudo apt-get --yes install --no-install-recommends libgbm1
       - restore_frontend_deps
       - restore_frontend_comm_deps
+      - restore_cache:
+          name: Restore Docker image from cache
+          key: << parameters.test_runner >>-image-v2-{{ .Environment.CIRCLE_WORKFLOW_WORKSPACE_ID }}
```

Comment:
Did you test whether this + `docker load` is faster than just pulling the pushed image from ECR? That of course incurs some costs but worth testing at least.

Reply:
The pull was slow but so is load. Tried it first and used this afterwards to speed up, but this did not speed up as mutch as I hoped. Did not make real comparision. If we want to use only local image as discussed above then we need to use this.

Reply:
:heavy_check_mark: I was hoping pulling would be way faster (considering how abysmally slow CircleCI's cache restores can be). But in that case, I think going full-local would be the best option. Maybe compare pull vs. cache restore speeds for that, too.